toffy/rename_fovs.py

```diff
@@ -0,0 +1,80 @@
+import json
+import os
+import warnings
+from distutils.dir_util import copy_tree
+
+from ark.utils import io_utils
+
+
+def rename_fov_dirs(run_path, fov_dir, new_dir=None):
+    """Renames FOV directories to have specific names sources from the run JSON file
+
+    Args:
+        run_path (str): path to the JSON run file
+        fov_dir (str): directory where the FOV subdirectories are stored
+        new_dir (str): path to the new directory to output files to, defaults to None
+
+    Raises:
+        KeyError: issue reading keys from the JSON file
+        ValueError: there are existing FOV directories that are not described the run file
+        UserWarning: not all fov names in the run file have an existing directory
+
+        """
+
+    io_utils.validate_paths(run_path)
+    io_utils.validate_paths(fov_dir)
+
+    #retieve FOV names and number of scans for each
+    with open(run_path) as f:
+        run_metadata = json.load(f)
+
+    fov_scan = dict()
+    for fov in run_metadata.get('fovs', ()):
+        name = fov.get('name')
+        run_order = fov.get('runOrder', -1)
+        scans = fov.get('scanCount', -1)
+        if run_order * scans < 0:
+            raise KeyError(f"Could not locate keys in {run_path}")
+
+        if scans > 1:
+            for scan in range(1, scans+1):
+                fov_name = f'fov-{run_order}-scan-{scan}'
+                fov_scan[fov_name] = f'{name}-{scan}'
+        else:
+            fov_name = f'fov-{run_order}-scan-{scans}'
+            fov_scan[fov_name] = name
+
+    #insert some kind of fov name validation
+
+
+    #retrieve the current FOV directory names
+    old_dirs = io_utils.list_folders(fov_dir, "fov")
+
+    #check that fovs & scan counts match the number of existing FOV directories
+    if not set(old_dirs).issubset(set(fov_scan.keys())):
```

Comment:
See if you can use the existing [verify](https://github.com/angelolab/ark-analysis/blob/4fcd44ce4e1edbbbf75fbaab69867bbedb39d89a/ark/utils/misc_utils.py#L146) functions for some of these

Reply:
verify_in_list() raises its own custom error printing the extra default_name directory that is not found in the json
![Screenshot (230)](https://user-images.githubusercontent.com/38049893/164367884-67de70ad-6323-48ad-a4de-f56413390b34.png)
so do we want to raise ValueError(f"FOV folders exceed the expected amount specified in {run_path}") on top of that to be more specific? or just leave it as it

Reply:
If you can name the variables passed to `verify` such that the error message is informative enough, then using that logic would be preferable. Otherwise, can do custom